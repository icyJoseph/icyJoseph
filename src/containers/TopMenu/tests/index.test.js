import React from "react";
import { shallow } from "enzyme";

import { TopMenu } from "../";
import { Menu } from "semantic-ui-react";

describe("TopMenu", () => {
  const props = {
    history: { push: jest.fn() }
  };
  const spy = jest.spyOn(TopMenu.prototype, "handleClick");
  const wrapper = shallow(<TopMenu {...props} />);

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has three children", () => {
    expect(wrapper.find(Menu.Item)).toHaveLength(3);
  });

  it("click handler pushes to history object", () => {
    wrapper.instance().handleClick(undefined, { name: "blog" });
    expect(spy).toHaveBeenCalled();
    expect(props.history.push).toHaveBeenCalledWith("/blog");
  });

  it("click handler pushes to history object when clicking home", () => {
    wrapper.instance().handleClick(undefined, { name: "home" });
    expect(spy).toHaveBeenCalled();
    expect(props.history.push).toHaveBeenCalledWith("/");
  });
});
