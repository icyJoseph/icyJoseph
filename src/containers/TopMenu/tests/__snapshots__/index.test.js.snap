// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMenu renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopMenu
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MenuItem
          active={true}
          name="home"
          onClick={[Function]}
/>,
        <MenuItem
          active={false}
          name="blog"
          onClick={[Function]}
/>,
        <MenuItem
          active={false}
          name="hacks"
          onClick={[Function]}
/>,
      ],
      "fluid": true,
      "inverted": true,
      "style": Object {
        "borderRadius": 0,
        "marginBottom": 0,
      },
      "widths": 3,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "name": "home",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "name": "blog",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "name": "hacks",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MenuItem
            active={true}
            name="home"
            onClick={[Function]}
/>,
          <MenuItem
            active={false}
            name="blog"
            onClick={[Function]}
/>,
          <MenuItem
            active={false}
            name="hacks"
            onClick={[Function]}
/>,
        ],
        "fluid": true,
        "inverted": true,
        "style": Object {
          "borderRadius": 0,
          "marginBottom": 0,
        },
        "widths": 3,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "name": "home",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "name": "blog",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "name": "hacks",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
